README.md

```diff
@@ -13,51 +13,57 @@ The followings are required for building Kilda controller:
  - Maven 3.3.9+
  - JDK8
- - Python 2.7+
  - Python 3.5+
  - Docker 19.03.3+
  - Docker Compose 1.20.0+
  - GNU Make 4.1+
  - Open vSwitch 2.9+
  
-For running virtual environment you additionally need linux kernel 4.18+ for OVS meters support
-
-On Ubuntu 18.04, you can install those dependencies like this:
 
+#### Dependency installation on Ubuntu 18.04 
+For running virtual environment you additionally need linux kernel 4.18+ for OVS meters support. A virtual environment 
+in that sentence means docker instance with open-vswitch in it. We use that for functional testing.
+```
+sudo apt install maven make openjdk-8-jdk openvswitch-switch python3-pip linux-generic-hwe-18.04
+```
 ```
-apt-get install maven openjdk-8-jdk python python3 docker.io docker-compose virtualenv make openvswitch-switch linux-generic-hwe-18.04 python3-setuptools python3-pip
+sudo pip3 install --upgrade pip
+sudo pip3 install tox docker-compose
 ```
+#### Dependency installation on Ubuntu 20.04
+
+```sudo apt install maven make openjdk-8-jdk openvswitch-switch python3-pip```
+
+```sudo pip3 install tox docker-compose```
 
 #### Gradle
 You can either install Gradle, or use Gradle wrapper:
- - Option 1 - Install Gradle (ensure that you have gradle 6.7 or later) - https://gradle.org/install/
-
- - Option 2 - Use Gradle wrapper. The Kilda repository contains an instance of Gradle Wrapper 
+ - Option 1 - Use Gradle wrapper. The Kilda repository contains an instance of Gradle Wrapper 
  which can be used straight from here without further installation.
+ - Option 2 - Install Gradle (ensure that you have gradle 6.7 or later) - https://gradle.org/install/
+
 
 #### Docker
 Note that your build user needs to be a member of the docker group for the build to work. 
 Do that by adding the user to /etc/groups and logging out and back in again.
 
+##### Basic installation instruction from Docker site
+
+```sudo apt-get install apt-transport-https ca-certificates curl gnupg lsb-release
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg
+echo "deb [arch=amd64 signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+sudo apt-get update
+sudo apt-get install docker-ce docker-ce-cli containerd.io
+sudo usermod -aG docker $USER
+# reboot or re-login for apply usermod command
+```
+
 #### Maven
 You also need to increase the maven RAM limit at least up to 1G.
 
 ```export MAVEN_OPTS="-Xmx1g -XX:MaxPermSize=128m"```
 
-#### Python
-Ensure that you have Python2 installed since some of build steps depends on it. 
-Possible option for that using virtual environment (`virtualenv`) with python interpreter version provided:
 
-```
-virtualenv --python=python2 .venv
-. .venv/bin/activate
-```
-
-Ensure that you have `tox` installed:
-```
-pip install -U pip
-pip install tox
-```
 #### Confd
```

Comment:
https://github.com/telstra/open-kilda/pull/4521 - since you removing confd, let's clean this up as well

Reply:
That PR for update instructions only, #4521 for removing confd. Why do you want to remove confd here?

Reply:
Because I already know about that pr, and don't want to update readme twice